Ingress2gateway crash on common.groupIngressPathsByMatchKey

<!--
Thank you for your interest in Gateway API! Please note that bug reports
here should only be used for bugs with the API itself, such as:

- Incomplete or inaccurate validation
- Problems with the API specification
- Flaws in conformance tests

Bugs that don't fit into one of the above categories should likely be filed
with the [implementation](https://gateway-api.sigs.k8s.io/implementations/)
of Gateway API that you're using.

Please use this template while reporting a bug and provide as much info as
possible. Not doing so may result in your bug not being addressed in a timely
manner. Thank you!
-->

I installed ingress2gateway via `go install`. I am running Fedora 43.

**What happened**:
ingress2gateway crashed with the following trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11afe56]

goroutine 1 [running]:
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common.groupIngressPathsByMatchKey({0xc00082eaa0, 0x1, 0xc0006bc8c0?})
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/providers/common/utils.go:158 +0x76
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common.(*ingressRuleGroup).toHTTPRoute(_, _, {_})
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/providers/common/converter.go:337 +0x71
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common.(*ingressAggregator).toHTTPRoutesAndGateways(0xc00005b280, {0x0?})
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/providers/common/converter.go:236 +0x654
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common.ToIR({0xc00063a000, 0x47, 0xc000b3e140?}, 0xc00054e0c0, {0xc000b3e144?})
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/providers/common/converter.go:53 +0x17e
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/ingressnginx.(*resourcesToIRConverter).convert(0xc000483350, 0xc0001352c0)
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/providers/ingressnginx/converter.go:47 +0x27e
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/ingressnginx.(*Provider).ToIR(0x1459a80?)
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/providers/ingressnginx/ingressnginx.go:62 +0x65
github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw.ToGatewayAPIResources({0x1840e70, 0x21c22c0}, {0xc0002aa170, 0xe}, {0x0, 0x0}, {0xc00070c520, 0x1, 0x1}, 0xc000112150)
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/pkg/i2gw/ingress2gateway.go:79 +0x459
github.com/kubernetes-sigs/ingress2gateway/cmd.(*PrintRunner).PrintGatewayAPIObjects(0xc0001f0d00, 0xc000229508, {0x0?, 0x0?, 0x0?})
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/cmd/print.go:91 +0x13f
github.com/spf13/cobra.(*Command).execute(0xc000229508, {0xc00069af60, 0x2, 0x2})
	/home/reinout/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xb02
github.com/spf13/cobra.(*Command).ExecuteC(0xc000229208)
	/home/reinout/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
	/home/reinout/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/kubernetes-sigs/ingress2gateway/cmd.Execute()
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/cmd/root.go:52 +0x70
main.main()
	/home/reinout/go/pkg/mod/github.com/kubernetes-sigs/ingress2gateway@v0.5.0/main.go:25 +0xf
```

**What you expected to happen**:
I expected a migrated configuration file that works with the Gateway API

**How to reproduce it (as minimally and precisely as possible)**:
I just run `ingress2gateway print --providers ingress-nginx` as per the documentation.

**Anything else we need to know?**:

--- a/pkg/i2gw/providers/common/utils.go
+++ b/pkg/i2gw/providers/common/utils.go
@@ -155,6 +155,9 @@ func groupIngressPathsByMatchKey(rules []ingressRule) orderedIngressPathsByMatch
 	}
 
 	for i, ir := range rules {
+		if ir.rule.HTTP == nil {
+			continue
+		}
 		for j, path := range ir.rule.HTTP.Paths {
 			ip := ingressPath{
 				ruleIdx:       i,

--- a/pkg/i2gw/providers/common/utils_test.go
+++ b/pkg/i2gw/providers/common/utils_test.go
@@ -44,6 +44,62 @@ func TestGroupIngressPathsByMatchKey(t *testing.T) {
 			},
 		},
 		{
+			name: "rule with nil http",
+			rules: []ingressRule{
+				{
+					rule: networkingv1.IngressRule{},
+				},
+				{
+					rule: networkingv1.IngressRule{
+						IngressRuleValue: networkingv1.IngressRuleValue{
+							HTTP: &networkingv1.HTTPIngressRuleValue{
+								Paths: []networkingv1.HTTPIngressPath{
+									{
+										Path:     "/test",
+										PathType: PtrTo(networkingv1.PathTypePrefix),
+										Backend: networkingv1.IngressBackend{
+											Service: &networkingv1.IngressServiceBackend{
+												Name: "test",
+												Port: networkingv1.ServiceBackendPort{
+													Number: 80,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: orderedIngressPathsByMatchKey{
+				keys: []pathMatchKey{
+					"Prefix//test",
+				},
+				data: map[pathMatchKey][]ingressPath{
+					"Prefix//test": {
+						{
+							ruleIdx:  1,
+							pathIdx:  0,
+							ruleType: "http",
+							path: networkingv1.HTTPIngressPath{
+								Path:     "/test",
+								PathType: &iPrefix,
+								Backend: networkingv1.IngressBackend{
+									Service: &networkingv1.IngressServiceBackend{
+										Name: "test",
+										Port: networkingv1.ServiceBackendPort{
+											Number: 80,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
 			name: "1 rule with 1 match",
 			rules: []ingressRule{
 				{

